### ISSUE TITLE
Strange crashes when using tuple index sets

### ISSUE THREAD

**Issue by hexahedria** [google-research/dex-lang#115]
Some things work with tuple index sets, but other things don't.

For instance, array creation works, but printing fails:
```
>=> arr : ((Fin 10) & (Fin 3)) => Real = for idx. 1.0

>=> :p arr
dex: Not implemented: %fst (0@((Fin 10)) , 0@((Fin 3)))
CallStack (from HasCallStack):
  error, called at src/lib/Interpreter.hs:80:8 in dex-0.1.0.0-HJSwFDIBGLApX0gaH8Fs9:Interpreter
```

`ordinal` works, but `fromOrdinal` fails
```
>=> (0 @ (Fin 2), 0 @ (Fin 3))
(0@(Fin 2), 0@(Fin 3))
>=> :t (0 @ (Fin 2), 0 @ (Fin 3))
((Fin 2) & (Fin 3))
>=> ordinal (0 @ (Fin 2), 0 @ (Fin 3))
0
>=> 0 @ (Fin 2 & Fin 3)
dex: Cannot convert IVar ((:>) (Name GenName "v" 2) (IValType IntType)) to TC (PairType (TC (IntRange (Con (Lit (IntLit 0))) (Con (Lit (IntLit 2))))) (TC (IntRange (Con (Lit (IntLit 0))) (Con (Lit (IntLit 3))))))
CallStack (from HasCallStack):
  error, called at src/lib/Imp.hs:384:19 in dex-0.1.0.0-HJSwFDIBGLApX0gaH8Fs9:Imp
```

**Comment by apaszke** [comment]
Ohh interesting. I might have broken some things with my recent changes. Shouldn't be too hard to fix though.

**Comment by apaszke** [comment]
Fixed in #118!
